addons/website_forum/models/forum_forum.py

```diff
@@ -131,11 +135,46 @@ def _get_default_welcome_message(self):
 
     can_moderate = fields.Boolean(string="Is a moderator", compute="_compute_can_moderate")
 
+    # tags
+    tag_ids = fields.One2many('forum.tag', 'forum_id', string='Tags')
+    tag_most_used_ids = fields.One2many('forum.tag', compute='_compute_tag_ids_usage')
+    tag_unused_ids = fields.One2many('forum.tag', compute='_compute_tag_ids_usage')
+
     @api.depends_context('uid')
     def _compute_can_moderate(self):
         for forum in self:
             forum.can_moderate = self.env.user.karma > forum.karma_moderate
 
+    @api.depends('post_ids', 'post_ids.tag_ids', 'post_ids.tag_ids.posts_count', 'tag_ids')
+    def _compute_tag_ids_usage(self):
+        if not self.tag_ids:
+            self.tag_unused_ids = self.tag_most_used_ids = False
+            return
+
+        tags_data = self.env['forum.tag'].search_read(
+            [('id', 'in', self.tag_ids.ids)],
+            fields=['id', 'forum_id', 'posts_count'],
+            order='forum_id, posts_count DESC, name, id',
+        )
+        current_forum_id = tags_data[0]['forum_id'][0]
+        current_most_used_count = 0
+        forum_tags = defaultdict(lambda: {'most_used_ids': [], 'unused_ids': []})
+
+        for tag_data in tags_data:
+            tag_id, tag_forum_id, posts_count = itemgetter('id', 'forum_id', 'posts_count')(tag_data)
+            if tag_forum_id[0] != current_forum_id:
+                current_most_used_count = 0
```

Comment:
this is annoying to do that in python :/ but looks like it's not possible with the `read_group` (we can't choose the order when we do `:array_agg`)

I also tried in SQL, but the query is quite complicated, so not sure it's worth it

So to resume we either need to
- use `read_group` but sort in python -> mhe
- do a SQL query (couldn't do something simple, so big mhe)
- current implementation (order in sql thanks to `search_read`, and manually group in python) -> ok for now

(I just add it here to keep a trace if we change our mind)
```py
self.tag_ids.flush_recordset()
self.env.cr.execute("""
    SELECT f.id AS forum_id,
           ARRAY_AGG(DISTINCT mu.id) FILTER (WHERE mu.id IS NOT NULL) AS most_used,
           ARRAY_AGG(DISTINCT uu.id) FILTER (WHERE uu.id IS NOT NULL) AS unused
      FROM forum_forum AS f
JOIN LATERAL (SELECT t.id
              FROM forum_tag AS t
             WHERE t.forum_id = f.id AND t.posts_count != 0
          ORDER BY t.posts_count DESC, name, id
             LIMIT %(most_used)s
           ) AS mu ON TRUE
 LEFT JOIN forum_tag AS uu
        ON uu.forum_id = f.id AND uu.posts_count = 0
     WHERE f.id = ANY(%(forum_ids)s)
  GROUP BY f.id
""", {'forum_ids': self.ids, 'most_used': MOST_USED_TAGS_COUNT})
result = {
    line['forum_id']: (line['most_used'], line['unused'])
    for line in self.env.cr.dictfetchall()
}

for forum in self:
    most_used, unused = result.get(forum.id, ([], []))
    forum.tag_most_used_ids = self.env['forum.tag'].browse(most_used)
    forum.tag_unused_ids = self.env['forum.tag'].browse(unused)
```